### DISCUSSION TITLE
Box management / description of stock at storage location in table?

### DISCUSSION THREAD

**Discussion by krisha** [Part-DB/Part-DB-server#488] [Ideas]
Hi,

first, this a very nice project - thanks! After a lot of research and testing I finally decided to use Part-DB.

For me having more than 100 sub-locations in a box did not make so much sense (big dropdowns etc.). So I used the description of the stock to use a row/column scheme. Unfortunately now I don't see any possibility to know what is in each slot of the box without diving in each component.

How do you do slot management in boxes?

Would it make sense to show the description of a stock also in the tables? Maybe configurable?
![stock description](https://github.com/Part-DB/Part-DB-server/assets/1581653/fbe6d08f-6309-4eba-88a8-6a7db3f0f5cf)

**Comment by bascuppen** [comment]
Personally I solved this by naming the storage location to something unique. To me, there is no added value in giving a storage location a 'generic' name, even if that location is nested into another location. It is true, when showing an overview of the parts, the parent location is only shown when hovering over the storage location. However, when viewing the part itself, the nested stock structure is shown, as well as the description of the stock location

![image](https://github.com/Part-DB/Part-DB-server/assets/27974586/18c904a2-46cd-4675-ab12-a5b5c848bd28)

If you dislike the big dropdown when selecting a stock location, it is even more important to have the names of the location as descriptive as possible: Everything unmutable should be in the name. So, if you have a cabinet on wheels, with drawers, organised in mini containers in rows an columns in said drawer, the location name should reflect the identifier of the mini container and could be C3D2R1C4. (cabinet 3, drawer 2, row 1, colum4). This way, when selecting stock locations in a drop down, users can easily filter the dropdown location results, with the added benefit to easily move that cabinet from parent location to another parent location. (from lab to main storage, to car, for example). If, in this example, you will be moving the drawers between cabinets, the container name should not contain the identifier of the cabinet, otherwise editing the location would mean an update of the name, as well as the parent location.
